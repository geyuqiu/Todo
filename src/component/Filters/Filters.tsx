import React from "react";

type TableProps = {
	title: string,
	completed: string,
	setTitle: Function,
	setCompleted: Function
}

export const Filters = ({title, completed, setTitle, setCompleted}: TableProps) => (
	<div>
		<input value={title} placeholder={'keyword...'}
		       onChange={(e: any) => {
			       setTitle(e.target.value || undefined);
		       }}
		/>
		<select
			value={completed}
			onChange={(e: any) => {
				setCompleted(e.target.value || undefined)
			}}
		>
			<option value="">-</option>
			{['yes', 'no'].map((option: string, i: number) => (
				<option key={i} value={option}>
					{option}
				</option>
			))}
		</select>
	</div>
)
