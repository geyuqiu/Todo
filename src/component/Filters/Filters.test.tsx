import {render} from "@testing-library/react";
import React from "react";

import {Filters} from "./Filters";

describe("Filters", () => {
	it("NoResults should match snapshot", () => {
		const {asFragment} = render(<Filters/>);
		expect(asFragment()).toMatchSnapshot();
	});
});
