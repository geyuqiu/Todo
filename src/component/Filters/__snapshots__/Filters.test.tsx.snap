// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters NoResults should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <input
      placeholder="keyword..."
      value=""
    />
    <select>
      <option
        value=""
      >
        -
      </option>
      <option
        value="yes"
      >
        yes
      </option>
      <option
        value="no"
      >
        no
      </option>
    </select>
  </div>
</DocumentFragment>
`;
